src/mem/abstract_mem.cc

```diff
@@ -447,10 +448,12 @@ AbstractMemory::access(PacketPtr pkt)
             pkt->setData(host_addr);
         }
         TRACE_PACKET(pkt->req->isInstFetch() ? "IFetch" : "Read");
-        stats.numReads[pkt->req->requestorId()]++;
-        stats.bytesRead[pkt->req->requestorId()] += pkt->getSize();
-        if (pkt->req->isInstFetch())
-            stats.bytesInstRead[pkt->req->requestorId()] += pkt->getSize();
+        if (collectStats) {
+            stats.numReads[pkt->req->requestorId()]++;
+            stats.bytesRead[pkt->req->requestorId()] += pkt->getSize();
+            if (pkt->req->isInstFetch())
+                stats.bytesInstRead[pkt->req->requestorId()] += pkt->getSize();
```

Comment:
It will reduce the likelihood of future bugs to use curly braces around this if statement (i.e., it's not clear that the curly brace on line 456 is for the outer if statement and not this one).